cpp/src/neighbors/cagra_c.cpp

```diff
@@ -38,12 +38,23 @@ void* _build(cuvsResources_t res, cuvsCagraIndexParams params, DLManagedTensor*
   auto res_ptr = reinterpret_cast<raft::resources*>(res);
   auto index   = new cuvs::neighbors::cagra::index<T, uint32_t>(*res_ptr);
 
-  auto build_params                      = cuvs::neighbors::cagra::index_params();
-  build_params.intermediate_graph_degree = params.intermediate_graph_degree;
-  build_params.graph_degree              = params.graph_degree;
-  build_params.build_algo =
-    static_cast<cuvs::neighbors::cagra::graph_build_algo>(params.build_algo);
-  build_params.nn_descent_niter = params.nn_descent_niter;
+  auto index_params                      = cuvs::neighbors::cagra::index_params();
+  index_params.intermediate_graph_degree = params.intermediate_graph_degree;
+  index_params.graph_degree              = params.graph_degree;
+
+  switch (params.build_algo) {
+    case cuvsCagraGraphBuildAlgo::IVF_PQ:
+      index_params.graph_build_params = cuvs::neighbors::cagra::graph_build_params::ivf_pq_params{};
+      break;
+    case cuvsCagraGraphBuildAlgo::NN_DESCENT:
+      cuvs::neighbors::cagra::graph_build_params::nn_descent_params nn_descent_params{};
```

Comment:
Can we consolidate the C++ and C structs so that we don't need to duplicate all of these values, please? It's going to be painful to manage two complete copies of these structs because changes are going to get made to one without being made to the other. 

I did this with the `DIstanceType` enum in `cuvs::distance`, for example. I think we should ultimately strive to write these once (in C++ or C) and then expose them through the other.